src/util/records.py

```diff
@@ -231,6 +229,18 @@ def get_trans_terms(value):
             )
 
 
+def _parse(arch, view_id):
+    try:
+        arch = arch.encode("utf-8") if isinstance(arch, unicode) else arch
+        return lxml.etree.fromstring(arch.replace(b"&#13;\n", b"\n"))
+    except lxml.etree.XMLSyntaxError as e:
+        if e.msg.startswith("Opening and ending tag mismatch"):
+            # this view is already wrong, we don't change it
+            _logger.warning("Skipping edit of invalid view (id=%s):\n%s", view_id, e.msg)
+            raise Skip()
+        raise
```

Comment:
Why move this to the `edit_view` method?
Now every consumer also needs to handle the `Skip` exception.
I would let the `XMLSyntaxError` bubble up.

Reply:
Well, what I said in the commit message, plus:
I did not find any caller handling `XMLSyntaxError` explicitly[^1] so far. So my feeling was that existing callers do not care what the exception is exactly, while new ones could make a distinction (or not) if it makes sense to them like it does for `adapt_domains` (or not). Maybe if we raise `Skip(str(e))`, the difference will be even smaller?

[^1]: `rg XMLSyntaxError` on all the scripts.

Reply:
@KangOl, after discussing with @aj-fuentes I went back to the simpler patch from the original PR.